doc/usr/source/2_input/1_lustre.rst

```diff
@@ -793,3 +793,155 @@ a linear one.
+define within the frame block. All variables defined within the frame block must be present in
+this list. Then, initial values are optionally specified for these variables. 
+Variables are defined within the frame block body (denoted by the ``let`` and ``tel`` kewords).
+It is possible to leave variables (partially or fully) undefined: On the first timestep, each variable
+is set equal to its initialization value, if one exists. On other timesteps, each undefined variable stutters 
+(it is set equal to its value on the previous timestep). 
+
+The following example involves three variables ``y1``, ``y2``, and ``y3``. Since ``y1`` is left
+undefined within the frame block body, it will always be equal to 0 (its initialization
+value). ``y2`` will have value ``100, 0, 1, 2, 3, ...`` because it is set equal to its initialization value (100)
+on the first timestep, but on other timesteps it is set equal to ``counter()``. Even though ``y3`` is fully 
+defined within the frame block (with no unguarded ``pre``s), its initialization value is still used, so it is equal
+to ``5, 1, 2, 3, ...``.
+
+.. code-block:: none
+
+   node example() returns (y1, y2, y3: int);
+   let
+      frame ( y1, y2, y3 )
+      (* Initializations *)
+      y1 = 0; y2 = 100; y3 = 5;
+
+      (* Body *)
+      let
+         y2 = pre counter();
+         y3 = counter();
+      tel
+   tel
+   
+      
+   node counter() returns (y: int);
+   let
+      y = 0 -> pre y + 1;
+   tel
+
+
+Frame conditions are especially useful when combined with the ``if`` statements described in the previous
+subsection, as variables can be left undefined in some branches of the ``if`` statement.
+
+.. code-block:: none
+
+   node example() returns (y1, y2: int);
+   let
+      frame ( y1, y2 )
+      (* Initializations *)
+      y1 = 0; 
+      y2 = 100;
+
+      (* Body *)
+      let
+         if (counter() < 10)
+         then
+            y1 = counter();
+         else
+            y2 = counter() * 2;
+         fi
+      tel
+   tel
+   
+      
+   node counter() returns (y: int);
+   let
+      y = 0 -> pre y + 1;
+   tel
+
+In the above example, ``y1`` is left undefined in the ``else`` branch of the ``if`` statement,
+and ``y2`` is left undefined in the ``then`` branch. ``y1`` is initialized on the first timestep,
+set to be equal to ``counter()`` on the second through tenth timesteps, and then stutters (staying at 9) for the 
+remaining timesteps. On the other hand, ``y2`` starts at its initialization value (100) and 
+stutters there for the first 10 timesteps, and then is set to ``counter() * 2`` for the remaining timesteps.
+
+Note that variables do not have to have initializations. For example, the following 
+code is supported because even though ``y2`` does not have an initialization, it is
+present in the list of variables ``frame ( y1, y2 )``.
+
+.. code-block:: none
+
+   frame ( y1, y2 )
+   y1 = init1;
+   let
+      y1 = expr1;
+      y2 = expr2;
+   tel
```

Comment:
I would clarify what is the initial value of a variable when it does not have an initialization (perhaps with a concrete example). Namely, it is the initial value of the expression assigned in the frame block body or undefined otherwise.